modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -137,14 +140,25 @@
 
     OverviewsController overviewsController;
 
+    private CoordinateReferenceSystem granuleCRS;
+
+    public GeneralEnvelope coverageEnvelope;
```

Comment:
The granule CRS is added and exposed via a getter, but this one is public?
The name is also confusing, what you really have is the granule/reader envelope, not the "coverage" one, indeed the descriptor does not deal with coverages at all.